### PR TITLE
Modernizing the codebase - removing unused imports

### PR DESCRIPTION
**Pull request by JanVogelsang**
This PR removes all unused imports in C/C++ files.

### CODE REVIEW

**Comment by JanVogelsang** [comment]
Can someone support me with the failing checks? Somehow no errors got reported and all checks pass, but still the pipeline exits with an error code of 1.

**Comment by heplesser** [comment]
> Can someone support me with the failing checks? Somehow no errors got reported and all checks pass, but still the pipeline exits with an error code of 1.

The problem are the compiler warnings:
```
Make                   | Failed                             |
[2600](https://github.com/nest/nest-simulator/actions/runs/3182211083/jobs/5187939958#step:9:2601)
|                        |                                    |
[2601](https://github.com/nest/nest-simulator/actions/runs/3182211083/jobs/5187939958#step:9:2602)
|                        | Errors  : 0                        |
[2602](https://github.com/nest/nest-simulator/actions/runs/3182211083/jobs/5187939958#step:9:2603)
|                        | Warnings: 4
```

Known warnings that are to be ignored are defined here

https://github.com/nest/nest-simulator/blob/e1c80d01f8318b5b2ce5d2f292cafb1f0e26047d/build_support/parse_build_log.py#L372-L376

with explicit line numbers. Since you removed one include line from `sli/scanner.cc`, the line numbers need to be updated.

**Comment by JanVogelsang** [comment]
There is a code inspection option for unused imports in CLion which is "clangd-based". However, no information is supplied how exactly the inspection is performed.

**Comment by jougs** [comment]
@JanVogelsang: Can you please fix the conflict? Thanks!

**Comment by JanVogelsang** [comment]
Well, one of the checks failed because of an unstable connection, as it seems. Is there any way for me to rerun the checks? The tests pass on my branch, so they will also pass here.

File: nestkernel/event_delivery_manager.cpp [hunk 1]
```diff
@@ -24,11 +24,7 @@
 
 // C++ includes:
 #include <algorithm> // rotate
-#include <iostream>
-#include <numeric> // accumulate
-
-// Includes from libnestutil:
-#include "logging.h"
+#include <numeric>   // accumulate
```
Comment (heplesser) [inline nestkernel/event_delivery_manager.cpp:27]: This belongs under C++ includes above. The header from libnestutil is `numerics.h`.

**Review by heplesser — CHANGES_REQUESTED** [review]
Just one little fix, see inline.

**Comment by jougs** [comment]
@clinssen, @pnbabu: Can you please check if this PR has consequences for NESTML? And this triggers a question: are `include`s in the NESTML templates added conditionally depending on what features are actually used in the model or is there just a bunch of default includes?

**Review by jougs — COMMENTED** [review]
Have these changes been suggested by a tool? If so, could this be made into a static code check that runs on the CI to reject commits that are adding unused `include`s? I'd be willing to wait for this PR until such a test is added.

**Comment by heplesser** [comment]
> Have these changes been suggested by a tool? If so, could this be made into a static code check that runs on the CI to reject commits that are adding unused `include`s? I'd be willing to wait for this PR until such a test is added.

Good point! Obviously, it needs to be a free open source tool.

**Comment by JanVogelsang** [comment]
The changes have been suggested by an built-in static-code analysis checker in CLion. They do not state which tools is used behind the scenes though.  
There is an open-source tool, which seems to do the same thing, which I have not used myself yet though: [include what you see](https://github.com/include-what-you-use/include-what-you-use)

File: nestkernel/event_delivery_manager.cpp (hunk 1 above)
Comment (JanVogelsang) [reply nestkernel/event_delivery_manager.cpp:27]: It actually is under C++ includes, see https://github.com/nest/nest-simulator/blob/28e42a40e6814fe248059be6c1a0e59b631555a1/nestkernel/event_delivery_manager.cpp#L27
It just looks like it's under libnestutil in the changelist, but the whole libnestutil block is actually gone.

Comment (heplesser) [reply nestkernel/event_delivery_manager.cpp:27]: Sorry, my oversight!

**Review by heplesser — APPROVED** [review]

**Review by jougs — APPROVED** [review]
@heplesser: Does your approval mean that we skip the addition of the static test for now? @JanVogelsang: can you please create an issue to remind us of the addition of a such a test?

I'm approving now, as I'm fine with the changes in this PR.

**Comment by heplesser** [comment]
@jougs Yes, I think it is better to get these changes in place and add the automated check later through a separate issue. I will merge now.

**Pull request merged**